server/webserver.go

```diff
@@ -23,6 +23,35 @@ import (
 	ffmpeg "github.com/livepeer/lpms/ffmpeg"
 )
 
+func (s *LivepeerServer) setServiceURI(serviceURI string) error {
+
+	parsedURI, err := url.Parse(serviceURI)
+	if err != nil {
+		glog.Error(err)
+		return err
+	}
+
+	glog.Infof("Storing service URI %v in service registry...", serviceURI)
+
+	tx, err := s.LivepeerNode.Eth.SetServiceURI(serviceURI)
+	if err != nil {
+		glog.Error(err)
+		return err
+	}
+
+	err = s.LivepeerNode.Eth.CheckTx(tx)
+	if err != nil {
+		glog.Error(err)
+		return err
+	}
+
+	// Avoids a restart if only the host has been changed.
+	// If the port has been changed, a restart is still needed.
+	s.LivepeerNode.ServiceURI = parsedURI
```

Comment:
Wouldn't the transcode server that is set up on boot up still be bound to the original `httpAddr`? https://github.com/livepeer/go-livepeer/blob/c18d0a77d6da7f3e31a4efe5f595111fc06944a1/server/rpc.go#L346